core/txpool/legacypool/tx_overflowpool.go

```diff
@@ -0,0 +1,171 @@
+package legacypool
+
+import (
+	"container/heap"
+	"fmt"
+	"sync"
+	"time"
+
+	"github.com/ethereum/go-ethereum/common"
+	"github.com/ethereum/go-ethereum/core/types"
+)
+
+// txHeapItem implements the Interface interface (https://pkg.go.dev/container/heap#Interface) of heap so that it can be heapified
+type txHeapItem struct {
+	tx        *types.Transaction
+	timestamp int64 // Unix timestamp (nanoseconds) of when the transaction was added
+	index     int
+}
+
+type txHeap []*txHeapItem
+
+func (h txHeap) Len() int { return len(h) }
+func (h txHeap) Less(i, j int) bool {
+	return h[i].timestamp < h[j].timestamp
+}
+func (h txHeap) Swap(i, j int) {
+	if i < 0 || j < 0 || i >= len(h) || j >= len(h) {
+		return // Silently fail if indices are out of bounds
+	}
+	h[i], h[j] = h[j], h[i]
+	if h[i] != nil {
+		h[i].index = i
+	}
+	if h[j] != nil {
+		h[j].index = j
+	}
+}
+
+func (h *txHeap) Push(x interface{}) {
+	item, ok := x.(*txHeapItem)
+	if !ok {
+		return
+	}
+	n := len(*h)
+	item.index = n
+	*h = append(*h, item)
+}
+
+func (h *txHeap) Pop() interface{} {
+	old := *h
+	n := len(old)
+	if n == 0 {
+		return nil // Return nil if the heap is empty
+	}
+	item := old[n-1]
+	old[n-1] = nil // avoid memory leak
+	*h = old[0 : n-1]
```

Comment:
It will pop the latest tx, right? but IMO, we wanna to pop the oldest tx.
Should we change it to:
```
	item := old[0]
	old[0] = nil // avoid memory leak
	*h = old[1 : n]
```

Reply:
> It will pop the latest tx, right? but IMO, we wanna to pop the oldest tx. So it is actually a Stack, not a Heap. Should we change it to:
> 
> ```
> 	item := old[0]
> 	old[0] = nil // avoid memory leak
> 	*h = old[1 : n]
> ```
> 
> But, it will update the all item's index...

heap.Pop will pop the oldest one:
heap.Pop will put the oldest item at the end firstly, compare by the func `Less`
then call txHeap.Pop

Reply:
ok, got it. but I wonder if it would cost lots of CPU resource for each Pop operation, as it needs to do sorting first